src/createAction.ts

```diff
@@ -44,6 +47,17 @@ export interface ActionMixin<T, O extends DoOptions<T>> {
 	 * Disable the task if enabled
 	 */
 	disable(): void;
+
+	/**
+	 * A method which may be called to configure the action after it's been created.
+	 *
+	 * It's up to the implementation to decide what happens if this method is called multiple times. Implementations
+	 * may throw or return a rejected promise.
+	 *
+	 * @param configuration The configuration. Implementations will need to cast to their expected configuration object
+	 * @return May return a promise in case configuration is asynchronous
+	 */
+	configure(configuration: Object): Promise<void> | void;
```

Comment:
Without an explicit index (something like `{ [ option: string ]: any }`), I think we will run into challenges with passing object literals... Also, should we consider a Generic like we do with the DoOptions?